### DISCUSSION TITLE
Is association file path not required to run run_tum_slam or run_tum_rgbd_slam ?

### DISCUSSION THREAD

**Discussion by Udaykiran87** [stella-cv/stella_vslam#134] [Q&A] [answered]
Hello Community,

Thank you for your great work.
I have few questions regarding slam using TUM RGBD dataset.
Question 1:
As per [this ](https://openvslam-community.readthedocs.io/en/latest/example.html#tum-rgbd-dataset), it says to use **run_tum_slam.cc** but example folder contains **run_tum_rgbd_slam.cc**. I assume it is a typo mistake.

Question 2:
As per [this](https://openvslam-community.readthedocs.io/en/latest/example.html#tum-rgbd-dataset) it says to use **association file path** as a command line input but run_tum_rgbd_slam.cc does not contain any such option in the code. As per official [rgbd-dataset](https://vision.in.tum.de/data/datasets/rgbd-dataset/tools#evaluation) depth and rgb images needs to be synchronized. Hence i am worried if **association file path** really required or not. Could you please clarify?

I guess we need to create new synchronized rgb.txt and depth.txt and then input their path as command line input to run_tum_rgbd_slam.cc.

Question 3:
I could see that there is no .cpp file available for **tum_rgbd_localization**. Is it safe to use run_video_localization by giving input as map created from the run_rgbd_slam.cc ? Do you have any plan to create **tum_rgbd_localization.cc**?

Thanks and Regards,
Udaykiran Patnaik.

**Comment by ymd-stella** [comment, marked as answer]
> Question 1:
As per this , it says to use run_tum_slam.cc but example folder contains run_tum_rgbd_slam.cc. I assume it is a typo mistake.

I think you're right.

> Question 2:
As per this it says to use association file path as a command line input but run_tum_rgbd_slam.cc does not contain any such option in the code. As per official rgbd-dataset depth and rgb images needs to be synchronized. Hence i am worried if association file path really required or not. Could you please clarify?
> 
> I guess we need to create new synchronized rgb.txt and depth.txt and then input their path as command line input to run_tum_rgbd_slam.cc.

I don't know because I have never used the RGBD dataset in a non-ROS setup. I would appreciate it if you could investigate.

> Question 3:
> I could see that there is no .cpp file available for tum_rgbd_localization. Is it safe to use run_video_localization by giving input as map created from the run_rgbd_slam.cc ?

No problem.

> Do you have any plan to create tum_rgbd_localization.cc?

There are no specific plans.

**Reply by Udaykiran87** [reply]
Regarding Question2:
I have used associate.py script to get the time synchronized  rgb.txt and depth.txt.  Please refer [this](https://vision.in.tum.de/data/datasets/rgbd-dataset/tools). I am then able to run slam using run_tum_rgbd_slam.cc.

Regarding Question 3:
I would try to contribute for tum_rgbd_localization.cc.
